internal/resource/url.go

```diff
@@ -332,6 +341,68 @@ func (f *Fetcher) fetchFromDataURL(u url.URL, dest io.Writer, opts FetchOptions)
 	return f.decompressCopyHashAndVerify(dest, bytes.NewBuffer(url.Data), opts)
 }
 
+// FetchFromAWSMetadata fetches metadata from the `IMDSv2` service if its configured, else
+// it will fallback to the `IMDSv1`. By default, the value of url is an IMDSv1 endpoint
+// i.e. http://169.254.169.254/2019-10-01/user-data.
+func (f *Fetcher) FetchFromAWSMetadata(u url.URL, opts FetchOptions) ([]byte, error) {
+	userdataLatestURL := url.URL{
+		Scheme: "http",
+		Host:   "169.254.169.254",
+		Path:   "latest",
+	}
+	token, err := f.fetchAWSIMDSV2Token(userdataLatestURL.String())
+	if err != nil && err != ErrIMDSV2 {
+		return nil, err
+	}
+	if err != nil && err == ErrIMDSV2 {
+		// Do nothing
+		f.Logger.Info("IMDSv2 service is disabled; fallback to IMDSv1")
+	}
+	if len(token) > 0 && err == nil {
```

Comment:
Is there a case where `err` is nil & the token is empty?

Reply:
It will fall back to the `v1` option